3rdparty/CEGUI/CMakeLists.txt

```diff
@@ -0,0 +1,73 @@
+if (IRR_BUILD_CEGUI)
+
+if (MSVC)
+    set(CEGUI_DEPS_URL "https://sourceforge.net/projects/crayzedsgui/files/CEGUI%20Mk-2%20Dependencies/0.8.x/cegui-deps-0.8.x-src.zip/download")
+    set(CEGUI_DEPS_ARCHIVE ${CMAKE_CURRENT_BINARY_DIR}/cegui-deps-0.8.x-src.zip)
+    set(CEGUI_DEPS_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cegui-cegui-dependencies-0ecdf3a9e49b)
+    set(CEGUI_DEPS_BUILD_PATH ${CEGUI_DEPS_PATH}/build)
+    set(CEGUI_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cegui)
+
+    if (NOT EXISTS "${CEGUI_DEPS_ARCHIVE}")
+        message(STATUS "Downloading CEGUI 0.8.x dependencies")
+        file(
+            DOWNLOAD "${CEGUI_DEPS_URL}" "${CEGUI_DEPS_ARCHIVE}"
+            SHOW_PROGRESS
+        )
+    endif()
+    message(STATUS "CEGUI 0.8.x downloaded")
+
+    if (NOT EXISTS "${CEGUI_DEPS_PATH}")
+        message(STATUS "Extracting CEGUI 0.8.x dependencies")
+        execute_process(
+            COMMAND ${CMAKE_COMMAND} -E tar xzf ${CEGUI_DEPS_ARCHIVE}
+            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+        )
+        message(STATUS "Extracting done")
+    endif()
+
+    message(STATUS "Building CEGUI 0.8.x dependencies")
+
+    execute_process(
+        COMMAND ${CMAKE_COMMAND} -E make_directory build
+        WORKING_DIRECTORY ${CEGUI_DEPS_PATH}
+    )
+
+    execute_process(
+        COMMAND ${CMAKE_COMMAND} ..
+          -DCEGUI_BUILD_CORONA=OFF
+          -DCEGUI_BUILD_DEVIL=OFF
+          -DCEGUI_BUILD_EXPAT=ON
+          -DCEGUI_BUILD_FREEIMAGE=OFF
+          -DCEGUI_BUILD_FREETYPE2=ON
+          -DCEGUI_BUILD_GLEW=OFF
+          -DCEGUI_BUILD_GLFW=OFF
+          -DCEGUI_BUILD_GLM=ON
+          -DCEGUI_BUILD_LUA=OFF
+          -DCEGUI_BUILD_MINIZIP=OFF
+          -DCEGUI_BUILD_PCRE=OFF
+          -DCEGUI_BUILD_SILLY=OFF
+          -DCEGUI_BUILD_TINYXML=OFF
+          -DCEGUI_BUILD_TOLUAPP=OFF
+          -DCEGUI_BUILD_XERCES=OFF
+          -DCEGUI_BUILD_ZLIB=OFF
+          -DCEGUI_MSVC_STATIC_RUNTIME=ON
+        WORKING_DIRECTORY ${CEGUI_DEPS_BUILD_PATH}
+    )
+
+    execute_process(
+        COMMAND ${CMAKE_COMMAND} --build ${CEGUI_DEPS_BUILD_PATH} --config Release -j 4
```

Comment:
I would use `$ENV{NUMBER_OF_PROCESSORS}` instead of hardcoding the number of jobs to 4.

`NUMBER_OF_PROCESSORS` (exists on Windows only) defines the number of available processors

Reply:
just make it a `max($ENV,8)` ;)

Reply:
thanks for suggestions, I will update

Reply:
updated